tests/test_batch.py

```diff
@@ -139,7 +137,10 @@ def test_batch_insert_context_manager_no_commit_on_error():
     except ValueError:
         assert len(col) == 0
         assert job1.status() == 'pending'
-        assert job1.result() is None
+        # TODO CHANGED Behavior: Result of jobs without a response fails on
```

Comment:
This is a slight change in the behavior.  I think it's a bit unclear for it to return None, especially since it would return None if return_result was false.  If the job was never sent to the server, I feel that calling result() on it should throw an error.  If you believe I'm wrong I can look for a way to revert this behavior.